File: Binary Tree Level Order Traversal.java
/**
 * Definition for binary tree
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        List<List<Integer>> result = new ArrayList<List<Integer>>();
        traverse(root, 0, result);
        return result;
    }
    private void traverse(TreeNode root, int level, List<List<Integer>> result) {
        if (root == null)
            return;
        while (result.size() <= level) {
            result.add(new ArrayList<Integer>());
        }
        result.get(level).add(root.val);
        traverse(root.left, level + 1, result);
        traverse(root.right, level + 1, result);
    }
}

File: Binary Tree Level Order Traversal.py
# Definition for a  binary tree node
# class TreeNode:
#	 def __init__(self, x):
#		 self.val = x
#		 self.left = None
#		 self.right = None

class Solution:
	def traverse(self, level, root, result):
		if (root == None):
			return
		while (len(result) < level + 1):
			result.append([])
		result[level].append(root.val)
		self.traverse(level + 1, root.left, result)
		self.traverse(level + 1, root.right, result)
	# @param root, a tree node
	# @return a list of lists of integers
	def levelOrder(self, root):
		level = 0
		result = []
		self.traverse(level, root, result)
		return result